crates/optimism/rpc/src/eth/transaction.rs

```diff
@@ -98,23 +102,66 @@ where
 #[derive(Clone, Debug, Copy)]
 pub struct OpTxBuilder;
```

Comment:
this type needs to hold the chain spec type, so that it can be accessed in call to `<OpTxBuilder as TransactionCompat>::fill` as `self.chain_spec`